calico/felix/test/test_config.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+felix.test.test_config
+~~~~~~~~~~~
+
+Top level tests for Felix configuration.
+"""
+import logging
+import mock
+import socket
+import sys
+from calico.felix.config import Config, ConfigException
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+import etcd
+real_client = etcd.client
+
+# Logger
+log = logging.getLogger(__name__)
+
+
+# Set of results from etcd.
+etcd_results = {}
+
+from collections import namedtuple
```

Comment:
I’d make a few changes to these module imports:

* There are module imports mixed with program flow – I’d consolidate them all at the top of the script.
* I’d remove `import mock`, since it doesn’t seem to be used.
* Maybe this is just personal preference, but I’d use the `try ... except ImportError` idiom for unittest. I don’t like that we have to import the sys module just to import another module.